Sources/Noora/Validator/ValidationResult.swift

```diff
@@ -0,0 +1,9 @@
+enum ValidationResult {
+    /// Indicates that the validation was successful.
+    case valid
+
+    /// Indicates that the validation failed with a list of errors.
+    ///
+    /// - Parameter errors: An array of validation errors.
+    case invalid(errors: [ValidatableError])
+}
```

Comment:
What about using [`Result<Void, ValidationError>` ](https://developer.apple.com/documentation/swift/result) which captures the two cases that we are trying to represent here?

Reply:
Nice catch! I've implemented a new error type that handles all validation errors: 

```swift
/// A custom error type that aggregates multiple validation errors.
struct ValidationError: Error {
    /// An array of validation errors that occurred.
    let errors: [ValidatableError]
}
```

And I return this from the validation method: `Result<Void, ValidationError>`.